handlers/analytics/log_processor.go

```diff
@@ -1,212 +1,162 @@
-	var metrics strings.Builder
-	now := time.Now().UnixMilli()
-	for k, v := range p.logs {
-		metrics.WriteString(p.toViewCountMetric(k, v, now))
-		metrics.WriteString(p.toRebufferRatioMetric(k, v, now))
-		metrics.WriteString(p.toErrorRateMetric(k, v, now))
-	}
-
-	// send data
-	err := p.sendMetricsString(metrics.String())
-	if err != nil {
-		glog.Errorf("failed to send analytics logs, err=%v", err)
-	}
-
-	// clear map
-	p.logs = make(map[labelsKey]map[string]metricValue)
-}
-
-func (p *LogProcessor) toViewCountMetric(k labelsKey, v map[string]metricValue, nowMs int64) string {
-	value := fmt.Sprintf("%d", len(v))
-	return p.toMetric(k, "view_count", value, nowMs)
-}
-
-func (p *LogProcessor) toRebufferRatioMetric(k labelsKey, v map[string]metricValue, nowMs int64) string {
-	var count int
-	var sumRebufferRate float64
-	for _, mv := range v {
-		if mv.count > 0 {
-			count += 1
-			sumRebufferRate += mv.sumRebufferRate / float64(mv.count)
+	var msgs []kafka.Message
+	for _, d := range p.logs {
+		key := []byte(d.SessionID)
+		value, err := json.Marshal(d)
+		if err != nil {
+			glog.Errorf("invalid analytics log event, cannot sent to Kafka, err=%v", err)
+			continue
 		}
+		msgs = append(msgs, kafka.Message{
+			Key:   key,
+			Value: value,
+		})
 	}
-	var rebufferRate float64
-	if count > 0 {
-		rebufferRate = sumRebufferRate / float64(count)
-	}
-	value := fmt.Sprintf("%f", rebufferRate)
-	return p.toMetric(k, "rebuffer_ratio", value, nowMs)
-}
+	p.logs = []LogData{}
 
-func (p *LogProcessor) toErrorRateMetric(k labelsKey, v map[string]metricValue, nowMs int64) string {
-	var count int
-	var sumErrorRatio float64
-	for _, mv := range v {
-		if mv.count > 0 {
-			count += 1
-			sumErrorRatio += mv.sumErrorRatio / float64(mv.count)
-		}
-	}
-	var errorRatio float64
-	if count > 0 {
-		errorRatio = sumErrorRatio / float64(count)
-	}
-	value := fmt.Sprintf("%f", errorRatio)
-	return p.toMetric(k, "error_rate", value, nowMs)
-}
-
-func (p *LogProcessor) toMetric(k labelsKey, name string, value string, nowMs int64) string {
-	return fmt.Sprintf(`%s{host="%s",user_id="%s",playback_id="%s",device_type="%s",browser="%s",country="%s"} %s %d`+"\n",
-		name,
-		p.host,
-		k.userID,
-		k.playbackID,
-		k.deviceType,
-		k.browser,
-		k.country,
-		value,
-		nowMs,
-	)
-}
-
-func (p *LogProcessor) sendMetricsString(metrics string) error {
-	client := &http.Client{Timeout: SendMetricsTimeout}
-	req, err := http.NewRequest("POST", p.promURL, bytes.NewBuffer([]byte(metrics)))
-	if err != nil {
-		return err
-	}
-	req.Header.Set("Content-Type", "text/plain")
-
-	resp, err := client.Do(req)
+	err := p.writer.WriteMessages(context.Background(), msgs...)
 	if err != nil {
-		return err
-	}
-	defer resp.Body.Close()
-
-	if resp.StatusCode != http.StatusOK && resp.StatusCode != http.StatusNoContent {
-		return fmt.Errorf("non-OK status code: %d", resp.StatusCode)
+		glog.Errorf("error while sending analytics log to Kafka, err=%v", err)
```

Comment:
Doesn't have to be in this PR, but this should probably be a metric that we can alert on

Reply:
Ok, created an XS task in Linear: https://linear.app/livepeer/issue/ENG-1730/add-metrics-and-alerting